lib/auth/auth_with_roles.go

```diff
@@ -2383,14 +2383,19 @@ func (a *ServerWithRoles) GetSessionRecordingConfig(ctx context.Context, opts ..
 }
 
 // SetSessionRecordingConfig sets session recording configuration.
-func (a *ServerWithRoles) SetSessionRecordingConfig(ctx context.Context, recConfig types.SessionRecordingConfig) error {
-	if err := a.action(apidefaults.Namespace, types.KindSessionRecordingConfig, types.VerbCreate); err != nil {
+func (a *ServerWithRoles) SetSessionRecordingConfig(ctx context.Context, newRecConfig types.SessionRecordingConfig) error {
+	storedRecConfig, err := a.authServer.GetSessionRecordingConfig(ctx)
+	if err != nil {
 		return trace.Wrap(err)
 	}
-	if err := a.action(apidefaults.Namespace, types.KindSessionRecordingConfig, types.VerbUpdate); err != nil {
-		return trace.Wrap(err)
+
+	for _, verb := range verbsToReplaceResourceWithOrigin(storedRecConfig) {
+		if err := a.action(apidefaults.Namespace, types.KindSessionRecordingConfig, verb); err != nil {
+			return trace.Wrap(err)
+		}
 	}
```

Comment:
What do you think about adding `(a *ServerWithRoles) actions(namespace string, resource string, actions ...string) error` for this use case? It would also be useful in other spots and the `actionOpts` don't really need to be included in it.
```suggestion
	if err := a.actions(apidefaults.Namespace, types.KindSessionRecordingConfig, verbsToReplaceResourceWithOrigin(storedRecConfig)); err != nil {
		return trace.Wrap(err)
	}
```

Reply:
I like this idea but I think such a helper would better be introduced in a standalone PR that should also clear up the current rather confusing situation with `VerbList` and `VerbRead` (the two RBAC verbs most commonly used in conjunction). Some "get multiple" operations require only `VerbList` (e.g. `GetSessions`) while others require both `VerbList` and `VerbRead` (e.g. `GetAuthServers`). 

Would you mind creating an issue for this?

Reply:
Created - https://github.com/gravitational/teleport/issues/7494
Interestingly, there are also some list methods that require `VerbRead` alone...